packages/flutter_tools/test/general.shard/base/error_handling_file_system_test.dart

```diff
@@ -91,6 +91,26 @@ void main() {
       when(mockFileSystem.path).thenReturn(MockPathContext());
     });
 
+    testWithoutContext('when access is denied', () async {
+      setupWriteMocks(
+        mockFileSystem: mockFileSystem,
+        fs: fs,
+        errorCode: 5,
+      );
+
+      final File file = fs.file('file');
+
+      const String expectedMessage = 'The flutter tool cannot access the file';
+      expect(() async => await file.writeAsBytes(<int>[0]),
```

Comment:
This test is good. A test that you get a tool exit for the operations that `_validateReadPermissions` is also needed.

Reply:
done